From stable-bounces@linux.kernel.org Thu Jul 19 00:27:57 2007
From: Stefan Richter <stefanr@s5r6.in-berlin.de>
Date: Thu, 19 Jul 2007 09:27:37 +0200 (CEST)
Subject: fw-ohci: fix "scheduling while atomic"
To: stable@kernel.org
Cc: Kristian Høgsberg <krh@redhat.com>, linux-kernel@vger.kernel.org
Message-ID: <tkrat.bf75c55e27c90af7@s5r6.in-berlin.de>
Content-Disposition: INLINE


From: Stefan Richter <stefanr@s5r6.in-berlin.de>

context_stop is called by bus_reset_tasklet, among else.

Fixes http://bugzilla.kernel.org/show_bug.cgi?id=8735.
Same as commit b980f5a224f3df6c884dbf5ae48797ce352ba139.

Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/firewire/fw-ohci.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/firewire/fw-ohci.c
+++ b/drivers/firewire/fw-ohci.c
@@ -586,7 +586,7 @@ static void context_stop(struct context 
 			break;
 
 		fw_notify("context_stop: still active (0x%08x)\n", reg);
-		msleep(1);
+		mdelay(1);
 	}
 }
 
